code/datums/status_effects/buffs.dm

```diff
@@ -99,6 +99,108 @@
+	id = "Hippocratic Oath"
+	status_type = STATUS_EFFECT_UNIQUE
+	duration = -1
+	tick_interval = 25
+	examine_text = "<span class='notice'>They seem to have an aura of healing and helpfulness about them.</span>"
+	alert_type = null
+	var/hand
+	var/deathTick = 0
+
+/datum/status_effect/hippocraticOath/on_apply()
+	//Makes the user passive, it's in their oath not to harm!
+	ADD_TRAIT(owner, TRAIT_PACIFISM, "hippocraticOath")
+	var/datum/atom_hud/H = huds[DATA_HUD_MEDICAL_ADVANCED]
+	H.add_hud_to(owner)
+	return ..()
+
+/datum/status_effect/hippocraticOath/on_remove()
+	REMOVE_TRAIT(owner, TRAIT_PACIFISM, "hippocraticOath")
+	var/datum/atom_hud/H = huds[DATA_HUD_MEDICAL_ADVANCED]
+	H.remove_hud_from(owner)
+
+/datum/status_effect/hippocraticOath/tick()
+	if(owner.stat == DEAD)
+		if(deathTick < 4)
+			deathTick += 1
+		else
+			owner.visible_message("<span class='notice'>[owner]'s soul is absorbed into the rod, relieving the previous snake of its duty.</span>")
+			var/mob/living/simple_animal/hostile/retaliate/poison/snake/healSnake = new(owner.loc)
+			var/list/chems = list("bicaridine", "perfluorodecalin", "kelotane")
+			healSnake.poison_type = pick(chems)
+			healSnake.name = "Asclepius's Snake"
+			healSnake.real_name = "Asclepius's Snake"
+			healSnake.desc = "A mystical snake previously trapped upon the Rod of Asclepius, now freed of its burden. Unlike the average snake, its bites contain chemicals with minor healing properties."
+			new /obj/effect/decal/cleanable/ash(owner.loc)
+			new /obj/item/rod_of_asclepius(owner.loc)
+			qdel(owner)
+	else
+		if(ishuman(owner))
+			var/mob/living/carbon/human/itemUser = owner
+			var/obj/item/heldItem = (hand ==  1 ? itemUser.l_hand : itemUser.r_hand)
+			if(!heldItem || !istype(heldItem, /obj/item/rod_of_asclepius)) //Checks to make sure the rod is still in their hand
+				var/obj/item/rod_of_asclepius/newRod = new(itemUser.loc)
+				newRod.activated()
+				if(hand)
+					if(itemUser.put_in_l_hand(newRod))
+						to_chat(itemUser, "<span class='notice'>The Rod of Asclepius suddenly grows back out of your arm!</span>")
+					else
+						if(!itemUser.has_organ("l_arm"))
+							new /obj/item/organ/external/arm(itemUser)
+						new /obj/item/organ/external/hand(itemUser)
+						itemUser.update_body()
+						itemUser.put_in_l_hand(newRod)
+						to_chat(itemUser, "<span class='notice'>Your arm suddenly grows back with the Rod of Asclepius still attached!</span>")
+				else
+					if(itemUser.put_in_r_hand(newRod))
+						to_chat(itemUser, "<span class='notice'>The Rod of Asclepius suddenly grows back out of your arm!</span>")
+					else
+						if(!itemUser.has_organ("r_arm"))
+							new /obj/item/organ/external/arm/right(itemUser)
+						new /obj/item/organ/external/hand/right(itemUser)
+						itemUser.update_body()
+						itemUser.put_in_r_hand(newRod)
+						to_chat(itemUser, "<span class='notice'>Your arm suddenly grows back with the Rod of Asclepius still attached!</span>")
```

Comment:
This code...I'm not fond of. Sadly, we don't have a way to get an item index to a particular hand and what bodypart that hand is index to, as well, so this is about the best we can do. If anyone has any code recommendations, let me know.